### PR TITLE
remove native-tls

### PR DESCRIPTION
**Pull request by 82marbag**
* Show how to plug a connector
* For TLS: keep rustls, only

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: rust-runtime/aws-smithy-client/src/hyper_ext.rs [hunk 1]
```diff
@@ -237,8 +228,8 @@ fn find_source<'a, E: Error + 'static>(err: &'a (dyn Error + 'static)) -> Option
 /// Builder for [`hyper_ext::Adapter`](Adapter)
 ///
 /// Unlike a Smithy client, the [`Service`] inside a [`hyper_ext::Adapter`](Adapter) is actually a service that
-/// accepts a `Uri` and returns a TCP stream. Two default implementations of this are provided, one
-/// that encrypts the stream with `rustls`, the other that encrypts the stream with `native-tls`.
+/// accepts a `Uri` and returns a TCP stream. One default implementation of this is provided,
+/// that encrypts the stream with `rustls`
```
Comment (ysaito1001) [inline rust-runtime/aws-smithy-client/src/hyper_ext.rs:232]:
```suggestion
/// that encrypts the stream with `rustls`.
```

File: rust-runtime/aws-smithy-client/Cargo.toml [hunk 2]
```diff
@@ -27,8 +25,8 @@ fastrand = "1.4.0"
 http = "0.2.3"
 http-body = "0.4.4"
 hyper = { version = "0.14.25", features = ["client", "http2", "http1", "tcp"], optional = true }
-# cargo does not support optional test dependencies, so to completely disable rustls when
-# the native-tls feature is enabled, we need to add the webpki-roots feature here.
+# cargo does not support optional test dependencies, so to completely disable rustls
+# we need to add the webpki-roots feature here.
```
[the remark this answers is not in the document]
Comment (82marbag) [reply rust-runtime/aws-smithy-client/Cargo.toml:29]: Done

**Review by hlbarber — APPROVED** [review]

File: design/src/transport/connector.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+The client had previously supported `native-tls`. You can use your custom connector like this.
+
+Create your connector:
+
+```rust
+/// A `hyper` connector that uses the `native-tls` crate for TLS. To use this in a smithy client,
+/// wrap it in a [hyper_ext::Adapter](crate::hyper_ext::Adapter).
+pub type NativeTls = hyper_tls::HttpsConnector<hyper::client::HttpConnector>;
+
+pub fn native_tls() -> NativeTls {
```
Comment (rcoh) [inline design/src/transport/connector.md:13]: we can do it post merge, but we should also include guidance for SDK customers

**Review by rcoh — APPROVED** [review]

**Review by crisidev — APPROVED** [review]

**Comment by stanal** [comment]
why remove native-tls? can give some suggestions , I'm fusing on which to use between native-tls and rustls .

**Comment by jdisanti** [comment]
@stanal - The native-tls crate doesn't allow us to configure cipher suites, so if a vulnerability is discovered in one of the ones it uses, we don't have any ability to address it ourselves. We also have a requirement that this list of cipher suites needs to be used:
https://github.com/awslabs/smithy-rs/blob/778244f801b720bb2b2325a648378f0a51d7ecc6/rust-runtime/aws-smithy-client/src/conns.rs#L30-L40

**Pull request merged**